Add addp command for adding tasks with priority

rosconap opened the pull request:
Adding `addp` to insert new tasks with priority. 

- Implement addp PRIORITY "TASK" syntax to add tasks with specified priority (A-Z)
- Priority is automatically converted to uppercase and formatted as (PRIORITY)
- Add comprehensive error handling for invalid priorities (numbers, special chars, multiple chars)
- Integrate with existing features like TODOTXT_DATE_ON_ADD and configuration options
- Update help documentation in shorthelp() and actionsHelp()
- Add complete documentation to USAGE.md with examples and usage patterns
- Add comprehensive test suite (t1050-addp.sh) with 23 test cases covering:
  - Basic functionality and priority validation
  - Case conversion (lowercase to uppercase)
  - Integration with projects and contexts
  - Error handling for all invalid input scenarios
  - Configuration integration (date on add, priority on add overrides)
  - Edge cases with special characters and quotes

Example usage:

```bash
  todo.sh addp A "High priority task"     # Creates: (A) High priority task
  todo.sh addp c "Medium priority task"   # Creates: (C) Medium priority task
```

**Before submitting a pull request,** please make sure the following is done:

- [x] Fork [the repository](https://github.com/todotxt/todo.txt-cli) and create your branch from `master`.
- [x] If you've added code that should be tested, add tests!
- [x] Ensure the test suite passes.
- [x] Lint your code with [ShellCheck](https://www.shellcheck.net/).
- [x] Include a human-readable description of what the pull request is trying to accomplish.
- [x] Steps for the reviewer(s) on how they can manually QA the changes.
- [ ] Have a `fixes #XX` reference to the issue that this pull request fixes.

rosconap committed:
Add addp command for adding tasks with priority

- Implement addp PRIORITY "TASK" syntax to add tasks with specified priority (A-Z)
- Priority is automatically converted to uppercase and formatted as (PRIORITY)
- Add comprehensive error handling for invalid priorities (numbers, special chars, multiple chars)
- Integrate with existing features like TODOTXT_DATE_ON_ADD and configuration options
- Update help documentation in shorthelp() and actionsHelp()
- Add complete documentation to USAGE.md with examples and usage patterns
- Add comprehensive test suite (t1050-addp.sh) with 23 test cases covering:
  - Basic functionality and priority validation
  - Case conversion (lowercase to uppercase)
  - Integration with projects and contexts
  - Error handling for all invalid input scenarios
  - Configuration integration (date on add, priority on add overrides)
  - Edge cases with special characters and quotes

Example usage:
  todo.sh addp A "High priority task"     # Creates: (A) High priority task
  todo.sh addp c "Medium priority task"   # Creates: (C) Medium priority task

inkarkat commented on tests/t1050-addp.sh:
```diff
@@ -0,0 +1,165 @@
+
+test_todo_session 'addp with date on add' <<EOF
+>>> todo.sh addp A "Task with date"
+1 (A) $(date '+%Y-%m-%d') Task with date
```
You can hard-code the date, it's fixed to the Unix epoch 1234500000 = 2009-02-13 in the test scripts ([here](https://github.com/todotxt/todo.txt-cli/blob/5d2727f4ec3bb5654ede97ab97617f0d6cef301b/tests/test-lib.sh?plain=1#L429)).

inkarkat commented on todo.sh:
```diff
@@ -45,6 +45,7 @@ shorthelp()
 		    add|a "THING I NEED TO DO +project @context"
 		    addm "THINGS I NEED TO DO
 		          MORE THINGS I NEED TO DO"
+		    addp PRIORITY "TEXT TO ADD"
```
Given that we have `add|a` and `pri|p`, I'd expect the new command to be available as `addpri|ap`; `addp` is an inconsistent mixture of long and short forms.

inkarkat commented on todo.sh:
```diff
@@ -1120,6 +1126,29 @@ case $action in
+    [ -z "$3" ] && die "$errmsg"
+    
+    # Validate priority is a single letter A-Z
+    if ! echo "$priority" | grep -q '^[A-Za-z]$'; then
```
We already have modern Bash-style checks ([here](https://github.com/todotxt/todo.txt-cli/blob/564ed3373de9495ee4f654879617ce4bfa322507/todo.sh?plain=1#L1420)); no need to use `grep`.

```suggestion
    if [[ "$priority" != @([A-Za-z]) ]]; then
```

inkarkat reviewed (commented):
This is a common add-on ([here's mine](https://github.com/inkarkat/todo.txt-cli-ex/blob/master/actions/addpri)), might as well be included in the core.

rosconap committed:
Address code review feedback: rename addp to addpri|ap and improve implementation

- Rename command from 'addp' to 'addpri|ap' to match existing patterns (add|a, pri|p)
- Replace grep-based priority validation with modern Bash syntax
- Update shorthelp() and actionsHelp() functions to show both command forms
- Update USAGE.md documentation to reflect new command name
- Replace dynamic date with hardcoded 2009-02-13 in tests per framework requirements
- Rename test file from t1050-addp.sh to t1050-addpri.sh for consistency
- Update all test references from addp to addpri throughout test suite

rosconap commented:
Updated with ecae92fc9744e3c9558bb99c065bf1017acd1b9d to address @inkarkat 's feedback. Shamefully i did not check out the add-on feature yet.

inkarkat commented on todo.sh:
```diff
@@ -45,6 +45,7 @@ shorthelp()
 		    add|a "THING I NEED TO DO +project @context"
 		    addm "THINGS I NEED TO DO
 		          MORE THINGS I NEED TO DO"
+		    addpri|ap PRIORITY "TEXT TO ADD"
```
```suggestion
		    addpri|ap PRIORITY "THING I NEED TO DO +project @context"
```

inkarkat commented on todo.sh:
```diff
@@ -177,6 +178,12 @@ actionsHelp()
 		      Adds SECOND THING I NEED TO DO to you todo.txt on its own line.
 		      Project and context notation optional.
 
+		    addpri PRIORITY "TEXT TO ADD"
+		    ap PRIORITY "TEXT TO ADD"
+		      Adds TEXT TO ADD to your todo.txt file on its own line with
```
```suggestion
		    addpri PRIORITY "THING I NEED TO DO +project @context"
		    ap PRIORITY "THING I NEED TO DO +project @context"
		      Adds THING I NEED TO DO to your todo.txt file on its own line with
```

inkarkat reviewed (approved):
Looks good! I'd only use the example task text from `add`, not from `addto`. The `TEXT TO ADD` isn't very specific (as it's not necessarily a new task when adding to `done.txt`).

cloudrkt commented on todo.sh:
```diff
@@ -177,6 +178,12 @@ actionsHelp()
 		      Adds SECOND THING I NEED TO DO to you todo.txt on its own line.
 		      Project and context notation optional.
 
+		    addpri PRIORITY "TEXT TO ADD"
+		    ap PRIORITY "TEXT TO ADD"
```
Implemented.

cloudrkt commented on todo.sh:
```diff
@@ -1120,6 +1127,29 @@ case $action in
+    [ -z "$3" ] && die "$errmsg"
+    
+    # Validate priority is a single letter A-Z
+    if [[ "$priority" != @([A-Za-z]) ]]; then
```
Implemented.
